### ISSUE TITLE
Common enchantment: use some static analyzer from time to time

### ISSUE THREAD

**Issue by Civil** [JACoders/OpenJK#787]
You can make use of clang's static analyzer. I'll post here link to an example report, once it'll finish to run. It's easy to run - you just need to do: 'CC=ccc-analyzer CXX=c++-analyzer cmake .' and build project as 'scan-build make -j4' (j can be changed). There'll be a lot of false-positives, but it *can* spot bugs.

There are also at least two proprietary static analyzers, which are free for Open Source projects:
1. Coverity: https://scan.coverity.com/users/sign_up
2. PVS-Studio: http://www.viva64.com (well, they are not advertising widly that they are free, but they got a blog post, from which at least from my understanding they can either check your project from time to time, or provide you with free analyzer)

**Comment by Civil** [comment]
clang's analyzer report:
https://drive.google.com/file/d/0B6t6k7KRJAULR2g1UGZiV1JEeDA/view?usp=sharing


Please note that it's 26MB Zip archive, and unpacked size is around 200M (report about 400 potential bugs).

**Comment by Civil** [comment]
If you are interested, I can try to go over clang's report and try to identify if this are actual bugs or not and fix some of them.

**Comment by Razish** [comment]
I believe we have Coverity, AppVeyor and a buildbot set up.

**Comment by ensiform** [comment]
Coverity isn't even being maintained and nobody else bothered to help dig into it further.  Plus there were issues that I couldn't figure out plus their support system wasn't letting me create proper account.

**Comment by Civil** [comment]
Well, I still think that at least clang's static analyzer will worth to look at. Especially because it's opensource.

**Comment by Razish** [comment]
Definitely, building occasionally with Clang's static analysis and asan is essential.

**Comment by ensiform** [comment]
I tried running coverity last night but it fails to pass because VS 2015 c++ runtime headers etc meaning it doesn't fully support 2015 and newer C++.

**Comment by Civil** [comment]
Well, I actually don't know much about Coverity, except they are free for opensource and claims to be one of the best on market.

**Comment by smcv** [comment]
> building occasionally with Clang's static analysis

After #741 lands, it should be easy to add a `scan-build` run to `.travis.yml` if you want?
